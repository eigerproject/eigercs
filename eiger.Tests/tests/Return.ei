func returnFive()
 ret 5
end
emitln(returnFive())

func addNumbers(a, b)
 ret a + b
end
emitln(addNumbers(4, 6))

func conditionalReturnOne(x)
 if x ?= 1 then ret 10
 else ret 20
end
end
emitln(conditionalReturnOne(1))

func conditionalReturnTwo(x)
 if x ?= 1 then ret 10
 else ret 20
end
end
emitln(conditionalReturnTwo(2))

func multiplyValues(a, b)
 ret a * b
end
func addAndMultiplyValues(a, b, c)
 let sumResult = a + b
 ret multiplyValues(sumResult, c)
end
emitln(addAndMultiplyValues(2, 3, 4))

func earlyReturnTest(x)
 if x ?= 1 then ret 10 
 emitln("This should not print")
 ret 20
end end
emitln(earlyReturnTest(1))

func outerOne()
 func innerOne(x)
 if x ?= 1 then ret 5 end
 ret 6
end
 ret innerOne(1)
end
emitln(outerOne())

func outerTwo()
 func innerTwo(x)
 if x ?= 1 then ret 5 end
 ret 6
end
 ret innerTwo(2)
end
emitln(outerTwo())
