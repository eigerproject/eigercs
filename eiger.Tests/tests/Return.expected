5
10
10
20
20
10
5
6